src/cata_tiles.cpp

```diff
@@ -1560,75 +1560,189 @@ bool cata_tiles::draw_from_id_string( std::string id, tripoint pos, int subtile,
                                             ll, apply_night_vision_goggles, height_3d );
 }
 
-bool cata_tiles::draw_from_id_string(std::string id, TILE_CATEGORY category,
-                                     const std::string &subcategory, tripoint pos,
-                                     int subtile, int rota, lit_level ll,
-                                     bool apply_night_vision_goggles, int &height_3d )
+const tile_type *cata_tiles::find_furniture_looks_like( std::string id )
+{
+    std::string looks_like = id;
+    int cnt = 0;
+    while( !looks_like.empty() && cnt < 10 ) {
+        const tile_type *lltt = find_tile_with_season( looks_like );
```

Comment:
`find_tile_with_season()` is invoked twice per `draw_from_id_string()`. The first invocation is [here](https://github.com/CleverRaven/Cataclysm-DDA/pull/24747/files#diff-6f8417614661f6a364e536794cfa57deR1695).

Reply:
I think the issue you're pointing out is addressed in  #24778

Reply:
I meant something else:

```c++
    const tile_type *tt = find_tile_with_season( id );	// <- 1st find_tile_with_season(id).
    
    if( !tt ) {
	...
        if( category == C_FURNITURE ) {
            tt = find_furniture_looks_like( id, effective_id );	// <- 2nd find_tile_with_season(id) with the same id inside the function
```

Map lookup times are noticeable in tiles, so this may slightly impact the overall performance.